examples/ex42.cpp

```diff
@@ -1034,18 +1119,44 @@ ApplyGradient3D(const int ne, const Array<double> &B_, const Array<double> &G_,
 
    MFEM_FORALL_3D(e, ne, q1d, q1d, q1d,
    {
+      // shared memory 1D Basis functions filled from b and g inputs
+      MFEM_SHARED tensor<double, q1d, d1d> B, G;
+      // shared memory placeholders for temporary contraction results
+      MFEM_SHARED tensor<double, 2, 3, q1d, q1d, q1d> smem;
       // cauchy stress
       MFEM_SHARED tensor<double, q1d, q1d, q1d, dim, dim> invJ_dsigma_detJw;
+      // du/dxi, ddu/dxi
+      MFEM_SHARED tensor<double, q1d, q1d, q1d, dim, dim> dudxi, ddudxi;
 
-      // du/dxi
-      tensor<double, q1d, q1d, q1d, dim, dim> dudxi{};
+      // initializer '{}' not allowed for shared variable
+      MFEM_FOREACH_THREAD(qx, x, q1d)
+      {
+         MFEM_FOREACH_THREAD(qy, y, q1d)
+         {
+            if (MFEM_THREAD_ID(z) == 0 && qy < d1d)
+            {
+               const int dy = qy;
+               B(qx, dy) = b(qx, dy);
+               G(qx, dy) = g(qx, dy);
+            }
```

Comment:
this seems like it it might be leaving part of `B`, `G` uninitialized if d1d > q1d

Reply:
the `d1d <= q1d` assumption should hold